### PR TITLE
Add foundational support for loading p5 addons in preview-frame.ts.

### PR DESCRIPTION
**Pull request by toolness**
This adds foundational support for loading multiple libraries which depend on p5.js in `preview-frame.ts`.  It effectively implements option 3 described in https://github.com/toolness/p5.js-widget/issues/53#issuecomment-240387563.

The biggest volatility here is that it uses an undocumented feature of the `p5` constructor whereby if a `sketch` function isn't passed in as the first argument, p5 will initialize itself in global mode.  If we can make this a documented, supported use case in p5 core, that will ensure the widget continues to work as p5 matures. I've filed https://github.com/processing/p5.js/issues/1570 for this.

Thoughts @kaganjd ?

### CODE REVIEW

File: lib/preview-frame.ts [hunk 1]
```diff
@@ -5,12 +5,16 @@ require("../css/preview-frame.css");
...
+  // even the first argument is actually *optional*; if omitted,
+  // p5 will initialize itself in global mode.
+  p5: (sketch?: Function, node?: HTMLElement, sync?: boolean) => void;
 }
 
 let global = window as PreviewFrameWindow;
 
-function loadP5(version: string, cb?: () => void) {
-  let url = '//cdnjs.cloudflare.com/ajax/libs/p5.js/' + version + '/p5.js';
+function loadScript(url, cb?: () => void) {
```
Comment (6a68) [inline lib/preview-frame.ts]: o hai :-) just a drive-by suggestion here: might be nice to listen for script load errors and log something to the console for debugging

File: lib/preview-frame.ts [hunk 2]
```diff
@@ -21,6 +25,24 @@ function loadP5(version: string, cb?: () => void) {
...
+      return cb();
+    }
+    loadScript(urls[i++], loadNextScript);
+  };
+
+  loadNextScript();
+}
+
+function p5url(version: string) {
+  return `//cdnjs.cloudflare.com/ajax/libs/p5.js/${version}/p5.js`;
```
Comment (6a68) [inline lib/preview-frame.ts]: might be worth it to allow a custom URL for self-hosting?

File: lib/preview-frame.ts [hunk 3]
```diff
@@ -21,6 +25,24 @@ function loadP5(version: string, cb?: () => void) {
   document.body.appendChild(script);
 }
 
+function loadScripts(urls: string[], cb?: () => void) {
```
Comment (6a68) [inline lib/preview-frame.ts]: :+1: you could get fancy with promises, but looping over the scripts and loading each should get the job done

File: lib/preview-frame.ts (hunk 1 above)
Comment (toolness) [inline lib/preview-frame.ts]: Ah good idea! I'll add that--this will also be useful for when we allow people to specify arbitrary URLs to third-party add-on libraries, since we want to provide good feedback if they misspell any of those URLs or something.

File: lib/preview-frame.ts (hunk 3 above)
Comment (toolness) [inline lib/preview-frame.ts]: Yeah, I really like promises, but I'm not actually sure if we include the polyfill in the preview-frame... and I'd like to keep the preview frame as light as possible in terms of external dependencies and stuff, hrm.  Lemme look into it!

File: lib/preview-frame.ts (hunk 2 above)
Comment (toolness) [inline lib/preview-frame.ts]: Yep, this is actually filed in #48 actually--I'd prefer to keep this PR nice and small, and add support for that in a future one. 😄

File: lib/preview-frame.ts (hunk 1 above)
Comment (6a68) [inline lib/preview-frame.ts]: Cool! Glad my drive-by review was helpful ^_^

**Comment by kaganjd** [comment]
@toolness Awesome! I'm still wrapping my head around the concept and syntax here, but I get that this lets us call p5 global mode after we're sure everything else has loaded. Seems more elegant than a bunch of iframes too. I'll stay tuned for the response to processing/p5.js#1570 :)
